### ISSUE TITLE
Volatile delivery

### ISSUE THREAD

**Issue by nichollsh** [FormingWorlds/PROTEUS#168]
We now have the initial components of an escape model integrated into PROTEUS. However, it is possible that accretion is also contemporaneous with magma ocean evolution. It would be interesting to explore what role this plays in the changing volatile inventory/distribution of young planets.

It should be relatively easy to "add" additional volatile mass to the inventory as the planet evolves, in the same manner in which escape removes it. We would need some estimate of the rate at which this occurs.

@timlichtenberg is a good idea and is there a nice way to estimate this?

**Comment by timlichtenberg** [comment]
Yes, having time-variable inventories of H-C-N-S together with fO2 is a good and important first step for accretion. Ultimately, we would need a varying (growing) size of the aragog grid to additional add ongoing mass accretion. The latter is something that would be highly valuable, but more a long-term idea and not first priority. The former (variable H-C-N-S + fO2) should be pretty straightforward to do I believe.

**Comment by timlichtenberg** [comment]
Regarding the estimation of mass accretion: this is pretty much unsolved in my view from a planet formation perspective. What we should do from a PROTEUS point-of-view is enabling arbitrary accretion paths (i.e. addition/loss of H-C-N-S and fO2) by read-in from somewhere else (a module, or data/text file that is produced from a different model).

**Comment by nichollsh** [comment]
Do we have something like an order of magnitude estimate for the duration and rate of accretion? 

Handling this with a module is a nice idea. Maybe this could even go into ZEPHYRUS?

**Comment by timlichtenberg** [comment]
- Timescale is ~1e6 (protoplanetary disk, addition by pebbles and/or planetesimals and/or nebular gass accretion) to ~1e8 years (big giant impacts, expoentially declining rate).
- The fractionation between H-C-N-S is potentially strongly fractionated; i.e., even for Earth it is 100% debated where each of them comes from, all from similar sources or others.
- The rate of growth (mass accretion) is possibly/likely correlated with the volatile addition rate; i.e, while a planet grows volatiles are added. But not *all* volatiles, perhaps just some.

Obviously I am making it complicated, but a simpler initial approach may be useful. One could, for example, just start with some volatile addition rate and fO2/time change and then add a splitting term for H-C-N-S fractionation.

**Comment by timlichtenberg** [comment]
The volatile accretion can be exponential or linear for example as a start.

**Comment by timlichtenberg** [comment]
And last comment: I would prefer a different module for accretion. Like escape, volatile and mass accretion is a near-boundless area of research in itself, and such a module would have substantial potential to grow massively over time.

**Comment by nichollsh** [comment]
Seems like there's a large number of ways in which to take this, so a separate model makes sense. Adding initial support into PROTEUS with a fixed rate (or choosing linear, exponential as you suggest) doesn't sound too tricky.

Now we just need someone to do this! I might have a look at making a stub module for this in the near future.

**Comment by nichollsh** [comment]
Created an initial stub repo here: https://github.com/FormingWorlds/ABASSI

**Comment by nichollsh** [comment]
See also this preprint https://arxiv.org/abs/2409.13820

**Comment by nichollsh** [comment]
https://www.nature.com/articles/s41561-021-00835-9

**Comment by timlichtenberg** [comment]
Well well well... 🙃
https://www.nature.com/articles/nature21045
https://iopscience.iop.org/article/10.3847/2041-8213/ac9521/meta
It's a complex and unresolved topic.

**Comment by nichollsh** [comment]
Another potential contribution here is the continuous delivery of material from the Solar wind. Can apparently be substantial in offsetting mass loss from escape: https://arxiv.org/abs/2512.24004
